commit 46f0e6984c84a0eaaa3c85a649245bb998124d76
Author: Arnaldo Carvalho de Melo <acme@redhat.com>
Date:   Fri Jan 4 15:10:00 2019 -0300

    perf test shell: Use a fallback to get the pathname in vfs_getname
    
    [ Upstream commit 03fa483821c0b4db7c2b1453d3332f397d82313f ]
    
    Some kernels, like 4.19.13-300.fc29.x86_64 in fedora 29, fail with the
    existing probe definition asking for the contents of result->name,
    working when we ask for the 'filename' variable instead, so add a
    fallback to that.
    
    Now those tests are back working on fedora 29 systems with that kernel:
    
      # perf test vfs_getname
      65: Use vfs_getname probe to get syscall args filenames   : Ok
      66: Add vfs_getname probe to get syscall args filenames   : Ok
      67: Check open filename arg using perf trace + vfs_getname: Ok
      #
    
    Cc: Adrian Hunter <adrian.hunter@intel.com>
    Cc: Jiri Olsa <jolsa@kernel.org>
    Cc: Namhyung Kim <namhyung@kernel.org>
    Link: https://lkml.kernel.org/n/tip-klt3n0i58dfqttveti09q3fi@git.kernel.org
    Signed-off-by: Arnaldo Carvalho de Melo <acme@redhat.com>
    Signed-off-by: Sasha Levin <sashal@kernel.org>

diff --git a/tools/perf/tests/shell/lib/probe_vfs_getname.sh b/tools/perf/tests/shell/lib/probe_vfs_getname.sh
index 1c16e56cd93e..7cb99b433888 100644
--- a/tools/perf/tests/shell/lib/probe_vfs_getname.sh
+++ b/tools/perf/tests/shell/lib/probe_vfs_getname.sh
@@ -13,7 +13,8 @@ add_probe_vfs_getname() {
 	local verbose=$1
 	if [ $had_vfs_getname -eq 1 ] ; then
 		line=$(perf probe -L getname_flags 2>&1 | egrep 'result.*=.*filename;' | sed -r 's/[[:space:]]+([[:digit:]]+)[[:space:]]+result->uptr.*/\1/')
-		perf probe $verbose "vfs_getname=getname_flags:${line} pathname=result->name:string"
+		perf probe -q       "vfs_getname=getname_flags:${line} pathname=result->name:string" || \
+		perf probe $verbose "vfs_getname=getname_flags:${line} pathname=filename:string"
 	fi
 }
 
diff -uN a/1.txt b/1.txt
--- a/dummy/rpi_1378_46f0e6984c84a0eaaa3c85a649245bb998124d76.txt	1970-01-01 00:00:00.000000000 +0000
+++ b/dummy/rpi_1378_46f0e6984c84a0eaaa3c85a649245bb998124d76.txt	2013-12-23 04:07:40.000000000 +0000
@@ -0,0 +1 @@
+dummy file to ensure patch has content.
